### PR TITLE
(#2910) Ensure Pester test file paths don't exceed Test Kitchen maximum lengths.

### LINKED ISSUE #2910: chocolatey-dummy-package nuspec file path too long when run on CCM Test Kitchen on Team City
### What You Are Seeing?

When running tests on Team City for CCM, the PushCommand tests fail to find the `chocolatey-dummy-package.nuspec` file. It seems to be that Test Kitchen fails to copy the file over to the test suite, and so the package is never generated.

The reason we do not see this issue locally with a test kitchen run is our image copies into a directory that is six characters shorter in length than where Team City copies to.

### What is Expected?

File is copied over as expected.

### How Did You Get This To Happen? (Steps to Reproduce)

1. Ran tests on Team City against CCM test suite with Tag set to `PushCommand`

### System Details

N/A.

### Output Log

Screenshot of the pertinent log from Team City run:

image

### PR DESCRIPTION
**Pull request by corbob**
## Description Of Changes

Add a test to ensure we don't exceed maximum path lengths in Test Kitchen.

## Motivation and Context

When run with some Test Kitchen configurations, files where the path exceeds about 120 characters from the `tests` directory don't get copied to the Test Environment. This results in failing tests due to missing files.

## Testing

1. Set the Max length variable in `Invoke-Tests.ps1` to `50`
2. Ran `vagrant up` from the `tests` directory
3. Confirmed that after building the script errored indicating files to shorten.

## Change Types Made

* [x] Bug fix (non-breaking change)
* [x] Tests update
* [ ] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)
* [ ] PowerShell code changes.

## Related Issue

* Fixes #2910 
* ENGTASKS-2346

## Change Checklist

* [ ] Requires a change to the documentation
* [ ] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [x] All new and existing tests passed.
* [ ] PowerShell v2 compatibility checked.

### CODE REVIEW

File: Invoke-Tests.ps1 [hunk 1]
```diff
@@ -26,6 +26,17 @@ param(
 )
 $packageRegex = 'chocolatey\.\d.*\.nupkg'
 
+# Check if there are any tests that exceed Test Kitchen maximum lengths
+$TestsLocation = Join-Path $PSScriptRoot tests
+$MaxFileNameLength = 110
+$LongFiles = Get-ChildItem $TestsLocation -Recurse | Where-Object { ($_.FullName.Length - $TestsLocation.Length) -gt $MaxFileNameLength } | ForEach-Object { [pscustomobject]@{ FullName = $_.FullName ; ReductionNeeded = $_.FullName.Length - $TestsLocation.Length - $MaxFileNameLength } }
```
Comment (vexx32) [inline Invoke-Tests.ps1:32]: Speaking of long lengths, this line is way too long :P 

```suggestion
$LongFiles = Get-ChildItem $TestsLocation -Recurse |
    Where-Object { ($_.FullName.Length - $TestsLocation.Length) -gt $MaxFileNameLength } |
    Select-Object -Property FullName, @{ Name = 'ReductionNeeded'; Expression = { $_.FullName.Length - $TestsLocation.Length - $MaxFileNameLength } }

```

File: Invoke-Tests.ps1 [hunk 2]
```diff
@@ -26,6 +26,17 @@ param(
 )
 $packageRegex = 'chocolatey\.\d.*\.nupkg'
 
+# Check if there are any tests that exceed Test Kitchen maximum lengths
+$TestsLocation = Join-Path $PSScriptRoot tests
+$MaxFileNameLength = 110
+$LongFiles = Get-ChildItem $TestsLocation -Recurse | Where-Object { ($_.FullName.Length - $TestsLocation.Length) -gt $MaxFileNameLength } | ForEach-Object { [pscustomobject]@{ FullName = $_.FullName ; ReductionNeeded = $_.FullName.Length - $TestsLocation.Length - $MaxFileNameLength } }
+
+if ($LongFiles) {
+    Write-Host "File paths too long for Test Kitchen use. Please shorten lengths:"
```
Comment (vexx32) [inline Invoke-Tests.ps1:35]:
```suggestion
    Write-Host "Tests' file paths may be too long for Test Kitchen use. Please shorten file names or paths:"
```

File: Invoke-Tests.ps1 [hunk 3]
```diff
@@ -26,6 +26,17 @@ param(
...
 $packageRegex = 'chocolatey\.\d.*\.nupkg'
 
+# Check if there are any tests that exceed Test Kitchen maximum lengths
+$TestsLocation = Join-Path $PSScriptRoot tests
+$MaxFileNameLength = 110
+$LongFiles = Get-ChildItem $TestsLocation -Recurse | Where-Object { ($_.FullName.Length - $TestsLocation.Length) -gt $MaxFileNameLength } | ForEach-Object { [pscustomobject]@{ FullName = $_.FullName ; ReductionNeeded = $_.FullName.Length - $TestsLocation.Length - $MaxFileNameLength } }
+
+if ($LongFiles) {
+    Write-Host "File paths too long for Test Kitchen use. Please shorten lengths:"
+    $LongFiles | ForEach-Object { Write-Host "$($_.ReductionNeeded), $($_.FullName)" }
```
Comment (vexx32) [inline Invoke-Tests.ps1:36]:
```suggestion
    $LongFiles | Out-String | Write-Host

```

**Review by vexx32 — CHANGES_REQUESTED** [review]
A few things we can maybe do a little more neatly here, but otherwise this is a good addition, thanks!

**Review by vexx32 — APPROVED** [review]
Looks good to me, thanks for adding this!
